plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/StandardProjectWizard.java

```diff
@@ -58,10 +57,26 @@ public boolean performFinish() {
       status = new Status(Status.ERROR, "todo plugin ID", errorCode, ex.getMessage(), null);
     }
     
-    // todo if fail, call setErrorMessage()
+    showErrorMessageIfNecessary(status);
+    
     return status.isOK();
   }
 
+  
+  // visible for testing
+  void showErrorMessageIfNecessary(IStatus status) {
+    // if fail, display error message in wizard
+    if (status == Status.CANCEL_STATUS) {
+      page.setErrorMessage("User canceled project creation");
+    } else if (!status.isOK()) {
+      String message = "Failed to create project";
+      if (status.getMessage() != null && !status.getMessage().isEmpty()) {
+        message += ": " + status.getMessage();
+      }
+      page.setErrorMessage(message);
```

Comment:
This doesn't seem right.  The page may not be the active page (for example, if we add more pages).  So at the very least this should use `getCurrentPage()` rather than `page`.  But I think for errors in this process it would be better to use the StatusManager:
```
StatusManager.getManager().show(status, StatusManager.SHOW);
```
That will pop up a dialog with the status details.

Alternatively log it.  My preference is via SLF4j (`org.slf4j.api` is pulled in both by Jetty and M2Eclipse, so is almost always present).